### ISSUE TITLE
zsh is not working (cd: string not in pwd)

### ISSUE THREAD

**Issue by sagor2364** [ohmyzsh/ohmyzsh#9354]
![image](https://user-images.githubusercontent.com/42068419/96346895-0e9f0580-10c0-11eb-8223-104123113294.png)

cd: string not in pwd
Please help me to solve my problem

-----

Transcript:
```console
$ cd "u:\My Space\C++\" && g++ Maincode.cpp -o Maincode && "u:\My Space\C++\"Maincode
cd: string not in pwd: u:\My Space\C++\" && g++ Maincode.cpp -o Maincode && "u:\My
```

**Comment by rwmitchell** [comment]
Why does your OS icon indicate Ubuntu but your path has ```u:```, suggesting dos instead?
You are also using backslashes to separate directories instead of the proper forward slashes.
You can see the proper slash in your prompt.

**Comment by sagor2364** [comment]
I am using windows 10, wsl and ubuntu that's why. Please help me to solve this problem

> Why does your OS icon indicate Ubuntu but your path has `u:`, suggesting dos instead?
> You are also using backslashes to separate directories instead of the proper forward slashes.
> You can see the proper slash in your prompt.

**Comment by rwmitchell** [comment]
I did.  Use the proper slash and then show how it failed.  There is no point in solving other problems until you at least fix the obvious stuff.

Next, before trying to build up the command, get each component working first, in this case, just the 'cd'.
Then, run g++.
When that works, you can combine the two to automate the process.

If you don't know how to change directories, you should probably put the compiler down.

**Comment by mcornella** [comment]
There are two problems here:

1. You're using a Windows path inside WSL, which is not allowed. Instead, to refer to Windows directories you need to use UNIX path conversion. WSL puts Windows units in `/mnt` by default (so `C:\` is in `/mnt/c`). For example, take a closer look to the path your prompt shows: `u:\My Space\C++` is `/mnt/u/My Space/C++`. That's what you'd use in `cd` instead:
   ```zsh
   cd "/mnt/u/My Space/C++"
   ```

2. This is an indirect effect of using Windows paths: you're using backslashes (`\`) in your path, which are used in UNIX to quote characters. So when you write `cd "u:\My Space\C++\"` the last backslash is quoting the last double quote, meaning that the string that started with the first double quote hasn't really finished yet, so the rest of the command is still in the first parameter to cd. Essentially, what cd is reading is
   ```zsh
   cd 'u:\My Space\C++\" && g++ Maincode.cpp -o Maincode && "u:\My' 'Space\C++\"Maincode'
   ```

   So, to clarify, `cd` is getting two parameters:
   1. `u:\My Space\C++\" && g++ Maincode.cpp -o Maincode && "u:\My`
   2. `Space\C++\"Maincode`.

   When cd gets two parameters it's essentially the same as `cd <old> <new>` (see [Zsh documentation](http://zsh.sourceforge.net/Doc/Release/Shell-Builtin-Commands.html#index-cd)):
   > The second form of cd substitutes the string **new** for the string **old** in the name of the current directory, and tries to change to this new directory.

**TL;DR**) so, to recap: point (2) is just an explanation of what's happening under the hood, you really only need to be concerned about point (1): **use UNIX path format (slashes `/` instead of backslashes `\`), and know that Windows units are by default under `/mnt` (so `U:\` is in `/mnt/u`).

I hope that's clear.
